File: programmers/lv0/코딩기초트레이닝/연산과조건문/조건문자열/Solution.go
package main

import (
	"fmt"
)

func solution(ineq string, eq string, n int, m int) int {

	judge := func(statement bool) int {
		if statement {
			return 1
		}
		return 0
	}

	switch ineq + eq {
	case "<=":
		return judge(n <= m)
	case "<!":
		return judge(n < m)
	case ">=":
		return judge(n >= m)
	case ">!":
		return judge(n > m)
	default:
		return -1
	}
}
func main() {
	fmt.Println(solution("<", "=", 20, 50))
	fmt.Println(solution(">", "!", 41, 78))
}

File: programmers/lv0/코딩기초트레이닝/연산과조건문/조건문자열/Solution.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

int solution(string ineq, string eq, int n, int m) {
    int answer = 0;
    string statement = ineq + eq;
    if (statement == "<=") answer = n <= m;
    else if (statement == "<!") answer = n < m;
    else if (statement == ">=") answer = n >= m;
    else if (statement == ">!") answer = n > m;
    else {
        throw logic_error("Invalid Statement");
    }
    return answer;
}

int main(int argc, char *argv[]) {
    cout << solution("<", "=", 20, 50) << endl;
    cout << solution(">", "!", 41, 78) << endl;
    cout << solution("<<", "!", 41, 78) << endl;
}